app/jobs/csv_export_job.rb

```diff
@@ -0,0 +1,85 @@
+require 'csv'
+
+class CsvExportJob < ActiveJob::Base
+  queue_as :csv_jobs
+ 
+  def perform(csv_export_id)
+    ActiveRecord::Base.connection_pool.with_connection do
+      csv_export = CsvExport.find(csv_export_id)
+      create_export_folder
+      export_task(csv_export)
+      cleanup_downloaded
+    end
+  end
+  
+  private
+  
+  def cleanup_downloaded
+    # clean up downloaded files older than 12 hours or any jobs stuck for over 1 month
+    # TODO delete files with no record associations
+    end_date = Time.now - Rails.application.config.samson.export_job.downloaded_age
+    timeout_date = Time.now - Rails.application.config.samson.export_job.max_age
+    @csv_exports = CsvExport.where("(status = 'downloaded' AND updated_at <= :end_date) OR updated_at <= :timeout_date",
+      {end_date: end_date, timeout_date: timeout_date})
+    if @csv_exports
+      @csv_exports.each do |csv_downloaded|
+        filename = get_filename(csv_downloaded)
+        File.delete(filename) unless !File.exist?(filename)
+        csv_downloaded.delete
+        Rails.logger.info("Downloaded file #{csv_downloaded.filename} deleted")
+      end
+    end
+  end
+  
+  def export_task(csv_export)
+    csv_export.filename
+    filename = get_filename(csv_export)
+
+    if csv_export.content == "deploys"
+      csv_export.started!
+      deploy_csv_export(filename)
+    else
+      csv_export.failed!
+    end
+
+    export_completed_notify(csv_export)
+
+    rescue
+
+    delete_file(get_filename(csv_export))
+    csv_export.failed!
+    Rails.logger.info("Export #{csv_export.filename} failed");
+  end
+  
+  def export_completed_notify(csv_export)
+    if csv_export.started?
+      CsvMailer.created_email(csv_export)
+      csv_export.finished!
+      Rails.logger.info("Export #{csv_export.filename} completed")
+    end
+  end
+  
+  def deploy_csv_export(filename)
+    @deploys = Deploy.joins(:stage).all
+    CSV.open(filename, 'w+') do |csv|
+      csv << ["Deploy Number", "Project Name", "Deploy Sumary", "Deploy Updated", "Deploy Created", "Deployer Name", "Buddy Name", "Production Flag", Deploy.joins(:stage).count.to_s + " Deploys"]
```

Comment:
It's to report number of records that should be in the report.  I couldn't find a better place for the information.  I guess it could be added to the very bottom of the report once all the records have been exported.  

The auditors care about how much rows match the condition in the system equals the number of rows in the report.